plugins/woocommerce/src/Internal/BatchProcessing/BatchProcessingController.php

```diff
@@ -408,4 +450,80 @@ protected function log_error( \Exception $error, BatchProcessorInterface $batch_
 
 		$this->logger->error( $error_message, array( 'exception' => $error ) );
 	}
+
+	/**
+	 * Determines whether a given processor is consistently failing based on how many recent consecutive failures it has had.
+	 *
+	 * @since 9.1.0
+	 *
+	 * @param BatchProcessorInterface $batch_processor The processor that we want to check.
+	 * @return boolean TRUE if processor is consistently failing. FALSE otherwise.
+	 */
+	private function is_consistently_failing( BatchProcessorInterface $batch_processor ): bool {
+		$process_details = $this->get_process_details( $batch_processor );
+		$max_attempts    = absint(
+			/**
+			 * Controls the failure threshold for batch processors. That is, the number of times we'll attempt to
+			 * process a batch that has resulted in a failure. Once above this threshold, the processor won't be
+			 * re-scheduled and will be removed from the queue.
+			 *
+			 * @since 9.1.0
+			 *
+			 * @param int $failure_threshold Maximum number of times for the processor to try processing a given batch.
+			 * @param BatchProcessorInterface $batch_processor The processor instance.
+			 * @param array $process_details Array with batch processor state.
+			 */
+			apply_filters(
+				'wc_batch_processing_attempts_threshold',
```

Comment:
Let's simplify the name to make it more explicit/obvious, what about `wc_batch_processing_max_attempts`?

Reply:
Renamed in 08088d27efff6bf61732627bde3a390c156395bf.